src/components/Setting/AddNetworkModal.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import {
+  Modal,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Button,
+  Input,
+} from '@nextui-org/react';
+
+interface AddNetworkModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onSubmit: (networkData: {
+    name: string;
+    chainId: number;
+    optimismPortalAddress: string;
+    rpcUrl: string;
+  }) => void;
+}
+
+const AddNetworkModal: React.FC<AddNetworkModalProps> = ({
+  isOpen,
+  onClose,
+  onSubmit,
+}) => {
+  const [formData, setFormData] = React.useState({
+    name: '',
+    chainId: '',
+    optimismPortalAddress: '',
+    rpcUrl: '',
+  });
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    onSubmit({
+      ...formData,
+      chainId: parseInt(formData.chainId),
+    });
+    setFormData({
+      name: '',
+      chainId: '',
+      optimismPortalAddress: '',
+      rpcUrl: '',
+    });
+    onClose();
+  };
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} size="lg">
+      <ModalContent>
+        <form onSubmit={handleSubmit}>
+          <ModalHeader className="font-normal text-2xl pt-4">
+            Add Network
+          </ModalHeader>
+          <ModalBody>
+            <div className="flex flex-col gap-4">
+              <Input
+                label="Network Name"
+                placeholder="e.g., Optimism"
+                value={formData.name}
+                onChange={(e) =>
+                  setFormData({ ...formData, name: e.target.value })
+                }
+                required
+              />
+              <Input
+                label="Chain ID"
+                placeholder="e.g., 10"
+                type="number"
+                value={formData.chainId}
+                onChange={(e) =>
+                  setFormData({ ...formData, chainId: e.target.value })
+                }
+                required
+              />
+              <div className="flex items-center gap-2 px-1">
+                <span className="text-sm text-default-500">Tech Stack:</span>
+                <span className="text-sm">OP Stack</span>
+              </div>
+              <Input
+                label="Portal Proxy Address"
+                placeholder="e.g., 0x..."
```

Comment:
nit: maybe fill in example contract address on Optimism so user can at least have a little clue about where to get this portal address. Though it is indeed quite challenging 😂
```suggestion
                placeholder="e.g., 0xbEb5Fc579115071764c7423A4f12eDde41f106Ed"
```